polylogue/cli_common.py

```
@@ -133,7 +147,29 @@ def choose_single_entry(
     if getattr(ui, "plain", False):
-        return entries[0], False
+        # Avoid blocking on stdin in non-interactive environments.
+        import sys
+
+        if not sys.stdin.isatty():
+            if hasattr(ui, "console"):
+                ui.console.print("[yellow]Plain mode cannot prompt; pass --all/IDs or rerun with --interactive.")
+            return None, True
+        if header:
+            ui.console.print(header)
+        for idx, line in enumerate(lines):
```

Ruff (0.15.9)

[warning] Loop control variable `idx` not used within loop body

(B007)